best_line.h

```diff
@@ -0,0 +1,52 @@
+#ifndef BEST_LINE_H
+#define BEST_LINE_H
+
+#include <float.h>
+#include <limits.h>
+
+#include <iostream>
+#include <unordered_map>
```

Comment:
hint for real life: use absl::flat_hash_map_instead, it is much faster.
ok for interview though.